app/Infrastructure/Controllers/PostOpenWalletController.php

```diff
@@ -24,7 +24,7 @@ public function __invoke(OpenWalletRequest $openWalletRequest, OpenWalletService
             ], Response::HTTP_NOT_FOUND);
         }
 
-        if ($walletId) {
+        if ($walletId != "Cache is full") {
```

Comment:
The successfull response is the most priority response of the controller. 
Change the order and set the successfull reponse to the last `response()->json`of the controller.

Reply:
```suggestion
        if (str_contains($walletId, 'Cache is full')) {
            return response()->json([
                'description' => 'cache is full',
            ], Response::HTTP_NOT_FOUND);
        }

        return response()->json([
            'description' => 'successful operation',
            'wallet_id' => str($walletId)
        ], Response::HTTP_OK);
```